qiskit/optimization/problems/optimization_problem.py

```diff
@@ -35,203 +41,188 @@ class OptimizationProblem:
     """
 
-    def __init__(self, *args):
+    def __init__(self, file_name: str = None):
         """Constructor of the OptimizationProblem class.
 
         The OptimizationProblem constructor accepts four types of argument lists.
 
-        op = qiskit.optimization.OptimizationProblem()
+        Args:
+            file_name: read a model from a file.
+
+        Examples:
+
+        >>> from qiskit.optimization import OptimizationProblem
+        >>> op = OptimizationProblem()
         op is a new problem with no data
 
-        op = qiskit.optimization.OptimizationProblem("filename")
+        >>> from qiskit.optimization import OptimizationProblem
+        >>> op = OptimizationProblem("filename")
         op is a new problem containing the data in filename.  If
         filename does not exist, an exception is raised.
 
         The OptimizationProblem object is a context manager and can be used, like so:
 
-        with qiskit.optimization.OptimizationProblem() as op:
-            # do stuff
-            op.solve()
+        >>> from qiskit.optimization import OptimizationProblem
+        >>> with OptimizationProblem() as op:
+        >>>     # do stuff
+        >>>     op.solve()
 
-        When the with block is finished, the end() method will be called
-        automatically.
+        When the with block is finished, the end() method will be called automatically.
         """
 
-        if len(args) > 1:
-            raise QiskitOptimizationError("Too many arguments to OptimizationProblem()")
-        self._disposed = False
-        self._name = None
+        self._name = ''
 
-        # see `qiskit.optimization.VariablesInterface()`
         self.variables = VariablesInterface()
 
-        # see `qiskit.optimization.LinearConstraintInterface()`
-        self.linear_constraints = LinearConstraintInterface(varindex=self.variables.get_indices)
+        # convert variable names into indices
+        varindex = self.variables.get_indices
 
-        # see `qiskit.optimization.QuadraticConstraintInterface()`
-        self.quadratic_constraints = QuadraticConstraintInterface(
-            varindex=self.variables.get_indices
-        )
-
-        # see `qiskit.optimization.ObjectiveInterface()`
-        # pylint: disable=unexpected-keyword-arg
-        self.objective = ObjectiveInterface(varindex=self.variables.get_indices)
-
-        # see `qiskit.optimization.SolutionInterface()`
+        self.linear_constraints = LinearConstraintInterface(varindex=varindex)
+        self.quadratic_constraints = QuadraticConstraintInterface(varindex=varindex)
+        self.objective = ObjectiveInterface(varindex=varindex)
         self.solution = SolutionInterface()
-
-        # see `qiskit.optimization.ProblemType()` -- essentially conversions from integers to
-        # strings and back
         self.problem_type = ProblemType()
-        self.my_problem_type = 0
+
+        # None means it will be detected automatically
+        self._problem_type = None
```

Comment:
`problem_type` is an `ProblemType` object that provide information of all problem types such as LP, QP, etc. It does not have prefix `_` because it is an CPLEX interface.
On the other hand, `_problem_type` is the problem type associated with this object. That's why it has prefix '_' to represent it as a class variable.

Reply:
`get_problem_type` has a docstrings to show an example.
```
        >>> from qiskit.optimization import OptimizationProblem
        >>> op = OptimizationProblem()
        >>> op.read("lpex.mps")
        >>> op.get_problem_type()
        0
        >>> op.problem_type[op.get_problem_type()]
        'LP'
```

Reply:
`get_problem_type` returns an integer `self._problem_type` representing the problem type of the problem.
`self.problem_type` converts the integer into a string 'LP' representing the problem type.

Reply:
This is how CPLEX does. So, we have to keep it although I prefer `Enum` instead of this style.

Reply:
Ok, this is a weird style, but there is probably a good reason behind this. Then let's keep it.